examples/funcalign/rsrm_synthetic_reconstruction.ipynb

```diff
@@ -0,0 +1,191 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## This example replicates the synthetic experiment in the paper\n",
+    "## \"Capturing Shared and Individual Information in fMRI Data\", J. Turek, C. Ellis, L. Skalaban, N. Turk-Browne, T. Willke"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Import some libraries that we will need"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import scipy.io\n",
+    "from scipy.stats import stats\n",
+    "import numpy as np\n",
+    "import brainiak.funcalign.srm as srm\n",
+    "import brainiak.funcalign.rsrm as rsrm\n",
+    "from mpl_toolkits.mplot3d import Axes3D\n",
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Experiment setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "voxels = 100\n",
+    "samples = 200\n",
+    "subjects = 10\n",
+    "features = 3\n",
+    "snr = 20 # in dB\n",
+    "amplitude = 8.0 # approximate amplitude used for the values in Si\n",
+    "p = 0.05 # probability of being non-zero in Si"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we create some synthetic data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create a Shared response R with K = 3\n",
+    "theta = np.linspace(-4 * np.pi, 4 * np.pi, samples)\n",
+    "z = np.linspace(-2, 2, samples)\n",
+    "r = z ** 2 + 1\n",
+    "x = r * np.sin(theta)\n",
+    "y = r * np.cos(theta)\n",
+    "curve = np.vstack((x, y, z))\n",
+    "print('Curve max, min values:', np.max(curve), np.min(curve))\n",
+    "\n",
+    "# Create the subjects' data\n",
+    "data = [None] * subjects\n",
+    "W = [None] * subjects\n",
+    "noise_level = 0.0\n",
+    "for s in range(subjects):\n",
+    "    R = curve\n",
+    "    W[s], _ = np.linalg.qr(np.random.randn(voxels, 3))\n",
+    "    data[s] = W[s].dot(R)\n",
+    "    noise_level += np.sum(np.abs(data[s])**2)\n",
+    "\n",
+    "# Compute noise_sigma from desired SNR\n",
+    "noise_level = noise_level / (10 ** (snr / 10))\n",
+    "noise_level = np.sqrt(noise_level  / subjects / voxels / samples)\n",
+    "\n",
+    "for s in range(subjects):\n",
+    "    n = noise_level * np.random.randn(voxels, samples)\n",
+    "    data[s] += n\n",
+    "    \n",
+    "S = [None] * subjects\n",
```

Comment:
Might be good to make another example where there is no S variability and show that RSRM does as well as SRM

Reply:
The idea of the example is to show how to use it. I chose the synthetic example as it is easy to replicate and visually show the improvements. If you set gamma to "infinity", you will be solving exactly the (deterministic) SRM.

Reply:
Good